src/overseer/core.clj

```diff
@@ -1,11 +1,88 @@
 (ns ^:no-doc overseer.core
   "Internal core functions"
   (:require [datomic.api :as d]
-            [clojure.set :as set]))
+            [clojure.set :as set]
+            [miner.herbert :as h]
+            loom.graph))
+
+(def job-schema
+  "Jobs are defined as maps with the requisite descriptive keys/values."
+  '{:job/id str
+    :job/type kw
+    :job/status (or :unstarted :started :finished :failed :aborted)
+    :job/heartbeat? int
```

Comment:
I think this is exactly the problem that `spec` mentioned about why it splits key definitions from different key sets.  Going _in_ `:job/heartbeat` is optional, but presumably in a variety of states after that it's not optional.  The "different sets and subsets of keys" problem.